cluster/swarm/cluster.go

```diff
@@ -170,6 +170,11 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 		config.SetSwarmID(swarmID)
 	}
 
+	if network := c.Networks().Get(config.HostConfig.NetworkMode); network != nil {
+		config.AddConstraint("node==~" + network.Engine.Name)
```

Comment:
I think it it so the container land on a machine where the network is.

I added a check to not add multiple constraints